### PR TITLE
Feature/ik streaming

### PR DESCRIPTION
**Pull request by Richiedu38**

### CODE REVIEW

**Review by SylvainBertrand — CHANGES_REQUESTED** [review]
Step 1: Format your code, I won't review non-formatted code :grin: 
Step 2: Go over the changes yourself and then ask me to review.

**Review by Richiedu38 — COMMENTED** [review]
formating

**Comment by Richiedu38** [comment]
Changes include assertions changes to run the velocity limit test, a new class to try to adjust null alpha in future experiments, changes in the construction of parameters to enable the velocity limitation and formatting on all of these.

File: ihmc-avatar-interfaces/src/main/java/us/ihmc/avatar/networkProcessor/kinematicsToolboxModule/KinematicsToolboxController.java [hunk 1]
```diff
@@ -781,7 +789,8 @@ protected boolean initializeInternal()
       else
       {
          initializePrivilegedConfiguration();
-         // It is required to update the tools now as it is only done at the end of each iteration.
+         // It is required to update the tools now as it is only done at the end of each
+         // iteration.
```
Comment (SylvainBertrand) [inline ihmc-avatar-interfaces/src/main/java/us/ihmc/avatar/networkProcessor/kinematicsToolboxModule/KinematicsToolboxController.java]: Normally the formatter isn't supposed to format one-line comments. We'll have to look at that and maybe do a revision of the formatter.

File: ihmc-avatar-interfaces/src/main/java/us/ihmc/avatar/networkProcessor/kinematicsToolboxModule/KinematicsToolboxPrivilegedConfigurationParameters.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    * uses this alpha parameter.
+    * <p>
+    * It should preferably be close to zero.
+    * </p>
+    * 
+    * @return the parameter used for computing the damped-least squares pseudo-inverse.
+    */
+   public double getNullspaceProjectionAlpha()
+   {
+      return 0.005;
```
Comment (SylvainBertrand) [inline ihmc-avatar-interfaces/src/main/java/us/ihmc/avatar/networkProcessor/kinematicsToolboxModule/KinematicsToolboxPrivilegedConfigurationParameters.java]: Copied-pasted code and doc and not the best. Here since it appears that the default values have not been changed, I'd suggest leaving this class empty, that'll simply redirect to the super class and won't change anything

File: ihmc-avatar-interfaces/src/main/java/us/ihmc/avatar/networkProcessor/kinemtaticsStreamingToolboxModule/KinematicsStreamingToolboxParameters.java [hunk 3]
```diff
@@ -14,34 +14,36 @@ public class KinematicsStreamingToolboxParameters
...
+       * Simple first-order estimation of the input velocities to then perform an extrapolation in the
+       * future using again first-order integration.
        */
       FIRST_ORDER_LPF,
       /**
-       * Slightly more complex estimation of the input velocities based on first order finite difference plus a correction based on the error in pose. This
-       * provides a continuous estimation of the input velocities.
+       * Slightly more complex estimation of the input velocities based on first order finite difference
+       * plus a correction based on the error in pose. This provides a continuous estimation of the input
+       * velocities.
```
Comment (SylvainBertrand) [inline ihmc-avatar-interfaces/src/main/java/us/ihmc/avatar/networkProcessor/kinemtaticsStreamingToolboxModule/KinematicsStreamingToolboxParameters.java]: Ok we really need to look at the formatter

File: ihmc-avatar-interfaces/src/main/java/us/ihmc/avatar/networkProcessor/kinemtaticsStreamingToolboxModule/KinematicsStreamingToolboxParameters.java [hunk 4]
```diff
@@ -293,7 +304,7 @@ public void setDefault()
       useStreamingPublisher = true;
       publishingPeriod = 5.0 * 0.006;
 
-      inputStateEstimatorType = InputStateEstimatorType.FIRST_ORDER_LPF;
+      inputStateEstimatorType = InputStateEstimatorType.FBC_STYLE;
```
Comment (SylvainBertrand) [inline ihmc-avatar-interfaces/src/main/java/us/ihmc/avatar/networkProcessor/kinemtaticsStreamingToolboxModule/KinematicsStreamingToolboxParameters.java]: The default value shouldn't be changed, I still want by default the old state estimator. Only change the estimator in your test and in `NadiaKinematicsStreamingToolboxModule`

**Review by SylvainBertrand — CHANGES_REQUESTED** [review]
I'm gonna take a look at the frigging formatter here. See if you can revert the change to the default value for the estimator

File: ihmc-avatar-interfaces/src/main/java/us/ihmc/avatar/networkProcessor/kinematicsToolboxModule/KinematicsToolboxPrivilegedConfigurationParameters.java [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+{
+<<<<<<< HEAD
+=======
+   public KinematicsToolboxPrivilegedConfigurationParameters()
+   {
+   }
+>>>>>>> branch 'feature/ik-streaming' of https://github.com/ihmcrobotics/ihmc-open-robotics-software.git
+
+<<<<<<< HEAD
+=======
```
Comment (SylvainBertrand) [inline ihmc-avatar-interfaces/src/main/java/us/ihmc/avatar/networkProcessor/kinematicsToolboxModule/KinematicsToolboxPrivilegedConfigurationParameters.java]: Something went wrong here :point_up:

**Review by SylvainBertrand — APPROVED** [review]
Nickel Michel!
